connectors/bigtable/README.md

```diff
@@ -0,0 +1,180 @@
+    // other settings
+    .build();
+```
+
+In this mode, all fields except `RowKeyField` must be Rows. The Row name represent the Column Family. For example:
+
+*   `key` (String)
+*   `family1` (contains fields like `name` (String), `age` (Integer))
+*   `family2` (contains fields like `city` (String), `zipCode` (Integer)) 
+
+Column Family `family1` would have columns `name, age` and Column Family `family2` would have columns `city, zipCode`.
+
+### Supported Types
+
+Both `RECORD` and `ROW` are not supported unless it's to define a Column Family in `Nested Rows Mode`. Double nested are not supported. For those and other unsupported types, you can use `BYTES`.
+
+#### `GenericRecordToRowMutationSerializer`
+
+The [Avro Types](https://avro.apache.org/docs/1.12.0/api/java/org/apache/avro/Schema.Type.html) supported are:
+
+* `STRING` 
+* `BYTES`  
+* `INT`  
+* `LONG` 
+* `FLOAT`  
+* `DOUBLE`  
+* `BOOLEAN`
+
+#### `RowDataToRowMutationSerializer`
+
+The [DataTypes](https://nightlies.apache.org/flink/flink-docs-release-1.19/api/java/index.html?org/apache/flink/table/api/DataTypes.html) supported are:
+
+* `STRING`
+* `CHAR`
+* `VARCHAR`
+* `BOOLEAN`
+* `TINYINT`
+* `SMALLINT`
+* `INT`
+* `BIGINT`
+* `DOUBLE`
+* `FLOAT`
+* `BYTES`
+* `BINARY`
+* `TIMESTAMP`
+* `TIMESTAMP_WITH_TIME_ZONE`
+* `TIMESTAMP_WITH_LOCAL_TIME_ZONE`
+* `INTERVAL(YEAR, MONTH)` 
+* `INTERVAL(DAY, SECOND)`
+* `TIME`
+* `DATE`
+* `DECIMAL` 
+
+The maximum precision for time-based types is 6.
+
+## Table API
+
+This connector provides support for Flink's Table API, enabling easy and efficient data writing to Bigtable tables within your Flink Table API pipelines. 
+
+The connector is named `bigtable`.
+
+### Row Key
+
+The Bigtable Flink Connector uses the primary key defined in your Flink schema as the row key for writing data to Bigtable. It must be of type `STRING` and cannot be null. Only one primary key can be used.
+
+```
+Schema schema =
+        Schema.newBuilder()
+                .column("my-key", DataTypes.STRING().notNull())
+                .column("stringColumn", DataTypes.STRING())
+                .column("intColumn", DataTypes.INT())
+                .primaryKey("my-key")
+                .build();
+```
+
+#### Connector Options
+
+The following connector options are available:
+
+| Option | Description |
+|---|---|
+| `project` | (Required) The Google Cloud project ID. |
+| `instance` | (Required) The Bigtable instance ID. |
+| `table` | (Required) The Bigtable table ID. |
+| `column-family` | The column family to write to (incompatible with `use-nested-rows-mode`). |
+| `use-nested-rows-mode` | Whether to use nested rows as column families (incompatible with `column-family`). |
+| `sink.parallelism` | The parallelism of the sink. |
+
+## Exactly Once
+
+Due to Bigtable's nature, the Sink offers Exactly Once out of the box. In order to get Exactly Once, every element needs to have a timestamp which will be used in the `RowMutationEntry`.
```

Comment:
have we confirmed this? ie, verified correctness / fault tolerance if TMs go down?

Reply:
Talked offline.

This was verified by checking the state of the reads by inserting data to Bigtable through the client libraries but not through the Sink itself.

Sink doesn't add any special recovery but relies on BT's idempotency + Flink's EO

Talking with the Bigtable team, this is right.

Anyhow, testing will be done